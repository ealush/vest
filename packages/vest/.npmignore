src
playground
config
testUtils
./jest.config.js
dist