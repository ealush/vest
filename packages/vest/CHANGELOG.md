# vest - Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 1.0.31 - 2021-07-02
### Fixed and improved
- e6ea7d6  support global replaces (undefined)
- 34e0414  improved conditions (undefined)
- 26c28c6  all tests pass (undefined)
- 33f4e46  release (undefined)
- 6fe40c7  better bundle (undefined)
- c2cfb65  better typing (undefined)
- c6387ab  before ts settings (undefined)
- c0e9708  generate correct d.ts file (undefined)
- packages/vest/types/vest.development.d.ts
- 8e01b8e  x (undefined)
- afb3960  x (undefined)
- e0a8463  add changelog support (undefined)
- cc46c38  current (undefined)
- 4a2073f  patch(vest): changes (ealush)
- c0a1705  patch(vest): changes (ealush)
- 240386c  better create generic (ealush)
- b6db1c6  transform any to unknowns (ealush)
- f2c20a7  add done integration tests (ealush)
- ba56e33  produce methods (ealush)
- 0638243  better produce integration tests (ealush)
- 4098a11  getFailures as integration tests (ealush)
- f56cf7f  pd (ealush)
- 2ea440e  pd (ealush)
- b3032ef  relocate draftresult (ealush)
- 962141f  test getFailuresByGroup (ealush)
- e163c56  test getFailures (ealush)
- a92b6df  test collectFailureMessages (ealush)
- 81aad51  fix most tests (ealush)
- c149e90  simplify collectFailureMessages (ealush)
