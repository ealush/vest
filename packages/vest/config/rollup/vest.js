const addEsmDir = require('../../../../config/rollup/addEsmDir');
const genConfig = require('../../../../config/rollup/genConfig');
const writeMainTemplate = require('../../../../config/rollup/writeMainTemplate');
const { packageDist, packageNames } = require('../../../../util');

addEsmDir(packageDist(packageNames.VEST));
writeMainTemplate(packageDist(packageNames.VEST), packageNames.VEST);

export default genConfig({
  distPath: packageDist(packageNames.VEST),
  esInput: 'vest.mjs.js',
  input: 'vest.js',
  libraryName: packageNames.VEST,
});
