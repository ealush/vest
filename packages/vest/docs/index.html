<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Vest - Validation Test</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="description" content="Validation Test" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    />
    <link rel="icon" href="./_assets/favicon.ico" />
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/docsify-themeable@0/dist/css/theme-simple.css"
    />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/prismjs@1.22.0/themes/prism-tomorrow.css">
  </head>
  <body>
    <div id="app"></div>
    <script>
      window.$docsify = {
        name: 'vest',
        logo: '/_assets/vest-logo.png',
        repo: 'https://github.com/ealush/vest',
        search: 'auto',
        loadSidebar: true,
        subMaxLevel: 2,
        themeColor: '#5BA9D8',
      };
    </script>
    <script src="//unpkg.com/docsify/lib/docsify.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/docsify-themeable@0"></script>
    <script src="//unpkg.com/docsify/lib/plugins/search.min.js"></script>
    <script src="//unpkg.com/vest"></script>
    <script src="./_assets/index.js"></script>
  </body>
</html>
