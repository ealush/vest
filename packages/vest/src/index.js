import any from "anyone/any";
import enforce from "n4s/dist/enforce.min";
import { VERSION } from "./constants";
import test from "./core/test";
import validate from "./core/validate";
import { draft, only, skip, warn } from "./hooks";
import { singleton, runWithContext } from "./lib";

export default singleton.register({
  Enforce: enforce.Enforce,
  VERSION,
  any,
  draft,
  enforce,
  only,
  runWithContext,
  skip,
  test,
  validate,
  warn,
});
