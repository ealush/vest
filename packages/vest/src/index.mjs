import enforce from '../../n4s/src/enforce';
import create from './core/suite/create';
import test from './core/test';
import { draft, only, skip, warn, group } from './hooks';

const VERSION = VEST_VERSION;

export default {
  VERSION,
  create,
  draft,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};

export { VERSION, create, draft, enforce, group, only, skip, test, warn };
