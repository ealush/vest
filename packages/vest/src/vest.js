import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
