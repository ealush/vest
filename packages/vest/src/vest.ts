import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
