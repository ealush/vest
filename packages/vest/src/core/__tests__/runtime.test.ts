import * as vest from 'vest';

describe('useLoadSuite', () => {
  it('Calling useLoadSuite should resume from loaded state', () => {
    const suite = vest.create(() => {
      vest.test('some_test', () => {});
    });
    const res = suite();
    expect(suite.isValid()).toBe(true);
    expect(res.testCount).toBe(1);
    expect(res.errorCount).toBe(0);
    expect(res.warnCount).toBe(0);
    expect(suite.hasErrors('some_test')).toBe(false);
    expect(suite.hasWarnings('some_test')).toBe(false);
    expect(suite.hasErrors()).toBe(false);
    expect(suite.hasWarnings()).toBe(false);
    expect(suite.getErrors()).toEqual({});

    const dump = genDump();

    expect(dump).toMatchSnapshot();

    suite.resume(dump);

    expect(res).not.toEqual(suite.get());
    expect(suite.isValid()).toBe(false);
    const loadedRes = suite.get();
    expect(loadedRes.testCount).toBe(6);
    expect(loadedRes.errorCount).toBe(4);
    expect(loadedRes.warnCount).toBe(1);
    expect(suite.hasErrors('t1')).toBe(true);
    expect(suite.hasErrors('t2')).toBe(true);
    expect(suite.hasErrors('t3')).toBe(false);
    expect(suite.hasErrors('t4')).toBe(false);
    expect(suite.hasErrors('t5')).toBe(false);
    expect(suite.hasErrors('t6')).toBe(true);
    expect(suite.hasWarnings('t1')).toBe(false);
    expect(suite.hasWarnings('t2')).toBe(false);
    expect(suite.hasWarnings('t3')).toBe(false);
    expect(suite.hasWarnings('t4')).toBe(true);
    expect(suite.hasWarnings('t5')).toBe(false);
    expect(suite.hasWarnings('t6')).toBe(false);
    expect(suite.hasErrors()).toBe(true);
    expect(suite.hasWarnings()).toBe(true);
    expect(suite.getErrors()).toEqual({
      t1: [],
      t2: ['t2 message'],
      t6: [],
    });
    expect(suite.hasErrorsByGroup('g1')).toBe(true);
    expect(suite.hasErrorsByGroup('g1', 't1')).toBe(false);
    expect(suite.hasErrorsByGroup('g1', 't2')).toBe(true);
    expect(suite.hasErrorsByGroup('g1', 't3')).toBe(false);
    expect(suite.hasErrorsByGroup('g1', 't4')).toBe(false);
    expect(suite.hasErrorsByGroup('g2')).toBe(false);
    expect(suite.hasErrorsByGroup('g2', 't1')).toBe(false);
    expect(suite.hasErrorsByGroup('g2', 't2')).toBe(false);
    expect(suite.hasErrorsByGroup('g2', 't3')).toBe(false);
    expect(suite.hasErrorsByGroup('g1', 't4')).toBe(false);
    expect(suite.hasWarningsByGroup('g1', 't4')).toBe(true);

    expect(suite.get()).toMatchSnapshot();
  });
});

function genDump() {
  const suite = vest.create(() => {
    vest.mode(vest.Modes.ALL);
    vest.skip('t5');

    vest.test('t1', () => false);
    vest.group('g1', () => {
      vest.test('t2', 't2 message', () => false);
      vest.test('t3', () => {});
      vest.test('t4', () => {
        vest.warn();
        return false;
      });
    });
    vest.test('t5', () => false);
    vest.each(['a', 'b'], key => {
      vest.test('t6', () => false, key);
    });
  });

  suite();

  return suite.dump();
}
