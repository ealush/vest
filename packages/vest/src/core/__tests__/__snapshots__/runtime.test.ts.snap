// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useLoadSuite Calling useLoadSuite should resume from loaded state 1`] = `
SuiteSummary {
  "errorCount": 4,
  "errors": [
    SummaryFailure {
      "fieldName": "t1",
      "groupName": undefined,
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "t2",
      "groupName": "g1",
      "message": "t2 message",
    },
    SummaryFailure {
      "fieldName": "t6",
      "groupName": undefined,
      "message": undefined,
    },
    SummaryFailure {
      "fieldName": "t6",
      "groupName": undefined,
      "message": undefined,
    },
  ],
  "getError": [Function],
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarning": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": {
    "g1": {
      "t2": SummaryBase {
        "errorCount": 1,
        "errors": [
          "t2 message",
        ],
        "pendingCount": 0,
        "testCount": 1,
        "valid": false,
        "warnCount": 0,
        "warnings": [],
      },
      "t3": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "pendingCount": 0,
        "testCount": 1,
        "valid": true,
        "warnCount": 0,
        "warnings": [],
      },
      "t4": SummaryBase {
        "errorCount": 0,
        "errors": [],
        "pendingCount": 0,
        "testCount": 1,
        "valid": true,
        "warnCount": 1,
        "warnings": [],
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "isValid": [Function],
  "isValidByGroup": [Function],
  "pendingCount": 0,
  "suiteName": undefined,
  "testCount": 6,
  "tests": {
    "t1": SummaryBase {
      "errorCount": 1,
      "errors": [],
      "pendingCount": 0,
      "testCount": 1,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "t2": SummaryBase {
      "errorCount": 1,
      "errors": [
        "t2 message",
      ],
      "pendingCount": 0,
      "testCount": 1,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "t3": SummaryBase {
      "errorCount": 0,
      "errors": [],
      "pendingCount": 0,
      "testCount": 1,
      "valid": true,
      "warnCount": 0,
      "warnings": [],
    },
    "t4": SummaryBase {
      "errorCount": 0,
      "errors": [],
      "pendingCount": 0,
      "testCount": 1,
      "valid": true,
      "warnCount": 1,
      "warnings": [],
    },
    "t5": SummaryBase {
      "errorCount": 0,
      "errors": [],
      "pendingCount": 0,
      "testCount": 0,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
    "t6": {
      "errorCount": 2,
      "errors": [],
      "pendingCount": 0,
      "testCount": 2,
      "valid": false,
      "warnCount": 0,
      "warnings": [],
    },
  },
  "valid": false,
  "warnCount": 1,
  "warnings": [
    SummaryFailure {
      "fieldName": "t4",
      "groupName": "g1",
      "message": undefined,
    },
  ],
}
`;
